### PR TITLE
Implemented refine result broadcast packet

### PR DESCRIPTION
**Pull request by secretdataz**
* **Addressed Issue(s)**: N/A
* **Server Mode**: Renewal
* **Description of Pull Request**: 
Implemented official refine announcement packet. Should work the way kRO works except name censoring which I don't know its logic.

### CODE REVIEW

**Comment by iberryRO** [comment]
WoW

File: src/map/status.cpp [hunk 1]
```diff
@@ -230,6 +230,32 @@ uint64 RefineDatabase::parseBodyNode( const ryml::NodeRef& node ){
 						}
 					}
 
+					if (this->nodeExists(refineLevelNode, "BroadcastSuccess")) {
+						bool bcast = false;
```
Comment (Lemongrass3110) [inline src/map/status.cpp:234]:
```suggestion
						bool bcast;
```

File: src/map/status.cpp [hunk 2]
```diff
@@ -230,6 +230,32 @@ uint64 RefineDatabase::parseBodyNode( const ryml::NodeRef& node ){
...
+						level_info->broadcast_success = bcast;
+					}
+					else {
+						if (!level_exists) {
+							level_info->broadcast_success = false;
+						}
+					}
+
+					if (this->nodeExists(refineLevelNode, "BroadcastFailure")) {
+						bool bcast = false;
```
Comment (Lemongrass3110) [inline src/map/status.cpp:247]:
```suggestion
						bool bcast;
```

File: src/map/clif.cpp [hunk 3]
```diff
@@ -24693,6 +24699,29 @@ void clif_parse_itempackage_select( int fd, struct map_session_data* sd ){
...
+
+	if (battle_config.broadcast_hide_name) {
+		std::string dispname = std::string(sd.status.name);
+		int hide = min(battle_config.broadcast_hide_name, dispname.length() - 1);
+		dispname.replace(dispname.length() - hide, hide, hide, '*');
+		safestrncpy(packet.name, dispname.c_str(), NAME_LENGTH);
+	}
+	else {
+		safestrncpy(packet.name, sd.status.name, NAME_LENGTH);
+	}
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:24719]: Can we put this in a separate function?
As this is used in clif_broadcast_obtain_special_item and here (and maybe should also be used in grade announce)?

File: src/map/clif.cpp [hunk 4]
```diff
@@ -24693,6 +24699,29 @@ void clif_parse_itempackage_select( int fd, struct map_session_data* sd ){
 #endif
 }
 
+void clif_broadcast_refine_result(map_session_data& sd, t_itemid itemId, int8 level, bool success)
+{
+#if PACKETVER_MAIN_NUM >= 20170906 || PACKETVER_RE_NUM >= 20170830 || defined(PACKETVER_ZERO)
+	PACKET_ZC_BROADCAST_ITEMREFINING_RESULT packet{};
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:24705]: Maybe rename from "packet" to "p" to match other functions.

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/clif.cpp (hunk 3 above)
Comment (secretdataz) [reply src/map/clif.cpp]: kRO does not hide the player's name. It could be an oversight, but I'm keeping it as is for now.

**Pull request merged**